aiohttp/web_urldispatcher.py

```diff
@@ -1052,20 +1113,36 @@ def register_resource(self, resource: AbstractResource) -> None:
                     "already handled by {!r}".format(name, self._named_resources[name])
                 )
             self._named_resources[name] = resource
-        self._resources.append(resource)
+        if append:
+            self._resources.append(resource)
 
     def add_resource(self, path: str, *, name: Optional[str] = None) -> Resource:
         if path and not path.startswith("/"):
             raise ValueError("path should be started with / or be empty")
+        # Reuse named resource
+        if name:
+            resource = self._named_resources.get(name)
+            if resource is None:
+                pass
+            elif resource.raw_match(path):
+                return cast(Resource, resource)
+            else:
+                raise ValueError(f"Duplicate {name!r}, already handled by {resource!r}")
+        if not ("{" in path or "}" in path or ROUTE_RE.search(path)):
+            if self._resources and isinstance(self._resources[-1], GroupPlainResource):
+                grp = self._resources[-1]
+            else:
+                grp = GroupPlainResource()
+                self.register_resource(grp)
+            resource = grp.add_resource(_requote_path(path), name=name)
+            if name:
+                self.register_resource(resource, append=False)
+            return resource
```

Comment:
It's not obvious to me why this has moved before the below block of code?

Reply:
To catch the last GroupPlainResource